syntax = "proto3";

package echo_journey.downward;

enum DownwardMessageType {
    UNKNOWN = 0;
    TUTOR_MESSAGE = 1;
    WORD_CORRECT_MESSAGE = 2;
    SENTENCE_CORRECT_MESSAGE = 3;
}

message DownwardMessage {
    DownwardMessageType type = 1;
    bytes payload = 2;
}

message TutorMessage {
    string text = 1;
    repeated WordCorrectMessage expected_messages = 2;
    bytes audio = 3;
}

message WordCorrectMessage {
    string word = 1;
    string initial_consonant = 2;
    string vowels = 3;
    int32 tone = 4;
    string pinyin = 5;
}

message CorrectMp4InfoMessage {
    string mp4_url = 1;
    string text = 2;
}

message SentenceCorrectMessage {
    repeated WordCorrectMessage expected_messages = 1;
    repeated WordCorrectMessage messages = 2;
    string suggestions = 3;
    float accuracy_score = 4;
    float fluency_score = 5; 
    repeated CorrectMp4InfoMessage correct_mp4_info = 6;
}